addon/services/activity-tracking.ts

```diff
@@ -0,0 +1,72 @@
+import Service, { inject as service } from '@ember/service';
+import Configuration from '@upfluence/hyperevents/configuration';
+
+export type ActivityType = 'page_view' | 'button_click' | 'component_view';
+export type Activity = {
+  type: ActivityType;
+  action?: string;
+  path: string;
+  route: string;
+  engine?: string;
+  version?: string;
+  extra?: {};
+};
+
+const RETRY_ATTEMPTS: number = 1;
+
+export default class ActivityTracking extends Service {
+  @service declare session: any;
+
+  log(activity: Activity): void {
+    // this method could also batch all the activities during XX seconds
```

Comment:
i think it'd be helpful here

Reply:
Let's discuss this when you arrive this afternoon. I see Pros & Cons to doing this :)